panic on v0.0.76

after an update to `v0.0.76` we're facing such panic quite often, which leads to the Pod crash
```
{"level":"info","msg":"promxy starting with HTTP...","time":"2023-03-16T12:20:25Z"}
panic: runtime error: index out of range [3] with length 3

goroutine 85351 [running]:
github.com/prometheus/prometheus/promql.findPathRange({0xc002562a80, 0x3, 0x24e9240?}, {0xc0013d2280?, 0x1, 0xc001347d78?})
	/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql/engine_extra.go:18 +0x19b
github.com/prometheus/prometheus/promql.(*Engine).findMinMaxTime.func1({0x38ed608?, 0xc00168ec60?}, {0xc002562a80?, 0x3, 0x3})
	/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql/engine.go:768 +0x11a
github.com/prometheus/prometheus/promql/parser.inspector.Visit(0xc000d88b40, {0x38ed608?, 0xc00168ec60?}, {0xc002562a80?, 0xc001347e70?, 0x4151ba?})
	/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql/parser/ast.go:383 +0x43
github.com/prometheus/prometheus/promql/parser.Walk({0x3900f00, 0xc00011a008}, {0x38e45a0, 0xc000d88b40}, 0xc001f696d0, {0x38ed608, 0xc00168ec60}, {0xc002562a80, 0x3, 0x3}, ...)
	/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql/parser/ast.go:330 +0x209
github.com/prometheus/prometheus/promql/parser.Walk.func1(0x0, {0x38ed608, 0xc00168ec60})
	/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql/parser/ast.go:344 +0x225
created by github.com/prometheus/prometheus/promql/parser.Walk
	/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/promql/parser/ast.go:342 +0x405
```
We're having such panic on every Pod in the Deployment, so it would be easy to know if it's fixed.
there are around 400 requests per second on this Promxy instance, fyi

--- a/Makefile
+++ b/Makefile
@@ -42,5 +42,5 @@ vendor:
 
 .PHONY: update-prom-fork
 update-prom-fork:
-	GO111MODULE=on $(GO) mod edit -replace github.com/prometheus/prometheus=github.com/jacksontj/prometheus@v0.2.37.3-fork
+	GO111MODULE=on $(GO) mod edit -replace github.com/prometheus/prometheus=github.com/jacksontj/prometheus@v0.2.37.4-fork
 	$(MAKE) vendor

--- a/go.mod
+++ b/go.mod
@@ -24,6 +24,7 @@ require (
 	golang.org/x/time v0.0.0-20220609170525-579cf78fd858
 	gopkg.in/yaml.v2 v2.4.0
 	k8s.io/klog v1.0.0
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
 )
 
 require (
@@ -158,13 +159,12 @@ require (
 	k8s.io/client-go v0.24.2 // indirect
 	k8s.io/klog/v2 v2.70.0 // indirect
 	k8s.io/kube-openapi v0.0.0-20220328201542-3ee0da9b0b42 // indirect
-	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
 	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace github.com/prometheus/prometheus => github.com/jacksontj/prometheus v1.8.1-0.20230309220703-968ded36c597
+replace github.com/prometheus/prometheus => github.com/jacksontj/prometheus v1.8.1-0.20230316162245-bf26a7edb634
 
 replace github.com/golang/glog => github.com/kubermatic/glog-gokit v0.0.0-20181129151237-8ab7e4c2d352
 

--- a/test/testdata/issue_562.test
+++ b/test/testdata/issue_562.test
@@ -1,0 +1,7 @@
+# Test for https://github.com/jacksontj/promxy/issues/562
+load 5m
+    metric_1{a="1"} 0+10x1000 100+30x1000
+
+eval instant at 5m max_over_time(metric_1[5m]) + metric_1
+    {a="1", az="a"} 20
+    {a="1", az="b"} 20

